src/scripts/main.js

```diff
@@ -1,3 +1,301 @@
+];
+
+form.classList.add('new-employee-form');
+formElements.forEach((element) => form.append(element));
+document.addEventListener('DOMContentLoaded', () => document.body.append(form));
+
+// FORM VALIDATION
+const validateFormInputData = (value, key) => {
+  const validators = {
+    position: () =>
+      !value
+        ? { validated: false, message: `Position is a required field!` }
+        : { validated: true },
+    name: () =>
+      !value || value.length < 4
+        ? { validated: false, message: 'Name should be more than 4 chars long' }
+        : { validated: true },
+    age: () =>
+      value < 18 || value > 90
+        ? { validated: false, message: 'Age must be between 18 and 90' }
+        : { validated: true },
+  };
+
+  return validators[key]?.() || { validated: true };
+};
+
+const validateForm = (data) => {
+  return Array.from(data)
+    .map(([key, value]) => validateFormInputData(value, key))
+    .filter((result) => !result.validated)
+    .map((result) => result.message);
+};
+
+// NOTIFICATIONS ON SUCESS / ERROR
+const displaySuccessNotice = (message) => {
+  displayNotification('success', 'Success!', [message]);
+};
+
+const displayErrorNotice = (errorArray) => {
+  displayNotification('error', 'Error!', errorArray);
+};
+
+const displayNotification = (type, titleText, errorMessageArr) => {
+  const notification = document.createElement('div');
+
+  notification.classList.add('notification', type);
+  notification.setAttribute('data-qa', 'notification');
+
+  notification.innerHTML = `
+    <h3 class="title">${titleText}</h3>
+    <p>
+      <ul>
+        ${errorMessageArr.map((message) => `<li>${message}</li>`).join('')}
+      </ul>
+    </p>
+  `;
+
+  document.body.append(notification);
+  setTimeout(() => notification.remove(), 2000);
+};
+
+// HANDLE SUBMIT EVENT
+const numToSalary = (num) => `$${(+num).toLocaleString('en-US')}`;
+const appendTableWithData = (data) => {
+  const newRow = tBody.insertRow(-1);
+
+  data.forEach((value, key) => {
+    newRow.insertCell(-1).textContent =
+      key === 'salary' ? numToSalary(value) : value;
+  });
+};
+
+const handleSubmitFormEvent = (e) => {
+  e.preventDefault();
+
+  const target = e.target.closest('form');
+  const formData = new FormData(target);
+  const errors = validateForm(formData);
+
+  if (errors.length === 0) {
+    const message = 'New employee has been added to the list!';
+
+    appendTableWithData(formData);
+    displaySuccessNotice(message);
+  } else {
+    displayErrorNotice(errors);
+  }
+};
+
+form.addEventListener('submit', handleSubmitFormEvent);
+
+// CELL EDITING EVENTS AND HANDLERS
+const createCellInput = (cell) => {
+  const input = Object.assign(document.createElement('input'), {
+    name: 'cellInput',
+    type: 'text',
+  });
+
+  input.classList.add('cell-input');
+  cell.append(input);
+
+  input.addEventListener('blur', handleSaveCellEvent);
+  input.addEventListener('keydown', handleEnterCellEvent);
+  input.focus();
+};
```

Comment:
Avoid creating elements and attaching event listeners inside a loop. It's better to create the element, attach the event listeners, and then append the element to the parent. Also, consider moving the `blur` and `keydown` event listeners to separate functions for better readability.